Mappy/MappingContext.cs

```diff
@@ -174,6 +174,17 @@ internal MappingContext(MappyOptions options, Abbrs abbreviations)
                 ?.AsList();
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal SortedSet<T> ConvertSortedSet<T>(
+            string prefix,
+            string name,
+            Items items,
+            List<Items> values,
+            IEnumerable<T> defaultValue)
+        {
+            return new SortedSet<T>(ConvertEnumerable(prefix, name, items, values, defaultValue) ?? Enumerable.Empty<T>());
+        }
+
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal IEnumerable<T> ConvertEnumerable<T>(
```

Comment:
To avoid unnecessary allocations we should rename `ConvertEnumerable` method to `ConvertCollection` and add `ICollection<T>` argument. Then we just pass `List<T>` in case of IEnumerable and `SortedSet<T>` in case of `SortedSet`.
Pay attention to [this ](https://github.com/Dolfik1/Mappy/blob/master/Mappy/MappingContext.cs#L202) line of code. We create `List` with predifined size to avoid allocations.

Reply:
I am not sure that I understand your thoughts. It seems like I don't understand something. What do I should change?
As far as I understand I need to rename existed `ConvertEnumerable `method to `ConvertCollection `method that will return ICollection<T>? What should I do after that?

Reply:
Have a look at the current code. You call `ConvertEnumerable` function which actually returns a `List<T>` casted to `IEnumerable<T>`. This leads to the memory allocation for the object `List<T>` which used only to create `SortedSet<T>`.
We can avoid unnecessary allocations if we pass instance of `SortedSet<T>` or `List<T>` object (depends on condition) as additional parameter of `ConvertEnumerable` function. So, we just remove this line `var result = new List<T>(values.Count);` from `ConvertEnumerable` function and add `result` parameter of `ICollection<T>` type.

Reply:
Done, only one note there is no constructor with count of elements for SortedSet, so we just create empty sorted set.